src/Microsoft.ML.TimeSeries/RootCauseLocalizationType.cs

```diff
@@ -0,0 +1,308 @@
+        /// </summary>
+        public RootCauseLocalizationInputTypeAttribute()
+        {
+        }
+
+        /// <summary>
+        /// Equal function.
+        /// </summary>
+        public override bool Equals(DataViewTypeAttribute other)
+        {
+            if (!(other is RootCauseLocalizationInputTypeAttribute otherAttribute))
+                return false;
+            return true;
+        }
+
+        /// <summary>
+        /// Produce the same hash code for all RootCauseLocalizationInputTypeAttribute.
+        /// </summary>
+        public override int GetHashCode()
+        {
+            return 0;
+        }
+
+        public override void Register()
+        {
+            DataViewTypeManager.Register(new RootCauseLocalizationInputDataViewType(), typeof(RootCauseLocalizationInput), this);
+        }
+    }
+
+    /// <summary>
+    /// Allows a member to be marked as a <see cref="RootCauseDataViewType"/>, primarily allowing one to set
+    /// root cause result.
+    /// </summary>
+    public sealed class RootCauseTypeAttribute : DataViewTypeAttribute
+    {
+        /// <summary>
+        /// Create an root cause type.
+        /// </summary>
+        public RootCauseTypeAttribute()
+        {
+        }
+
+        /// <summary>
+        /// RootCauseTypeAttribute with the same type should equal.
+        /// </summary>
+        public override bool Equals(DataViewTypeAttribute other)
+        {
+            if (other is RootCauseTypeAttribute otherAttribute)
+                return true;
+            return false;
+        }
+
+        /// <summary>
+        /// Produce the same hash code for all RootCauseTypeAttribute.
+        /// </summary>
+        public override int GetHashCode()
+        {
+            return 0;
+        }
+
+        public override void Register()
+        {
+            DataViewTypeManager.Register(new RootCauseDataViewType(), typeof(RootCause), this);
+        }
+    }
+
+    public sealed class RootCause
+    {
+        public List<RootCauseItem> Items { get; set; }
+        public RootCause()
+        {
+            Items = new List<RootCauseItem>();
+        }
+    }
+
+    public sealed class RootCauseLocalizationInput
+    {
+        //When the anomaly incident occurs
+        public DateTime AnomalyTimestamp { get; set; }
+
+        //Point with the anomaly dimension must exist in the slice list at the anomaly timestamp, or the libary will not calculate the root cause
+        public Dictionary<string, Object> AnomalyDimension { get; set; }
+
+        //A list of points at different timestamp. If the slices don't contain point data corresponding to the anomaly timestamp, the root cause localization alogorithm will not calculate the root cause as no information at the anomaly timestamp is provided.
+        public List<MetricSlice> Slices { get; set; }
+
+        //The aggregated symbol in the AnomalyDimension and point dimension should be consistent
```

Comment:
Why are they needed here then?

Reply:
> Why are they needed here then?

If the aggregation type is provided, some value can be obtained according to the aggration type when score is calcualted. For example , for sum type, when the detected root cause is not provided, we can simulate a point and caclulate the score